uxarray/core/dataarray.py

```diff
@@ -97,3 +96,112 @@ def uxgrid(self):
     @uxgrid.setter
     def uxgrid(self, ugrid_obj):
         self._uxgrid = ugrid_obj
+
+    def to_geodataframe(self,
+                        override_grid=False,
+                        cache_grid=True,
+                        correct_antimeridian_polygons=True):
+        """Constructs a ``spatialpandas.GeoDataFrame`` with a "geometry"
+        column, containing a collection of Shapely Polygons or MultiPolygons
+        representing the geometry of the unstructured grid, and a data column
+        representing a 1D slice of data mapped to each Polygon.
+
+        Parameters
+        override_grid : bool
+            Flag to recompute the ``GeoDataFrame`` if one is already cached
+        cache_grid : bool
```

Comment:
I'll update the docstring to better describe what's happening.

`cache_grid` determines whether the geometry (i.e. GeoDataFrame or PolyCollection) that is construct rued should be re-used for subsequent calls. For example, if the user constructs as `GeoDataFrame` with one data variable and then would like to use the same grid to map another data variable, `cache_grid` is set to True by default to avoid uncessary re-runs of `Grid.to_geodataframe` or `Grid.to_polycollection`

`override_grid` is used to determine whether the user wants to re-compute the grid instead of using a cached version.

Reply:
Thanks! Actually the "_grid" in each variable can get confusing; we are not caching or overriding the actual `grid` object but the geodataframe or polycollection that is being generated from that object. We should consider using a better name